webpack/config/webpack.production.config.js

```diff
@@ -0,0 +1,32 @@
+const webpack = require('webpack');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const WebpackBuildNotifierPlugin = require('webpack-build-notifier');
+const PATHS = require('../PATHS');
+
+const config = {
+  module: {
+    loaders: [
+       {
+         test: /\.scss$/,
+         loader: ExtractTextPlugin.extract('style', 'css', 'sass'),
+         include: PATHS.style
+       }
+     ]
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      'process.env': {
+        'NODE_ENV': JSON.stringify('production'),
```

Comment:
This is really interesting, DefinePlugin can be used as a dependency injection tool, it allows us to pass free variables or constants to our bundle, it replaces the matches "as is", thats why we are passing the String in this format, every module that references process.env.NODE_ENV inside our bundle will get a String ‘production’ instead, if we don’t put quotes around it, it would refer to a 'development' object which probably would be undefined inside that module, I used JSON.stringify to try to make that clearer.

`new webpack.DefinePlugin({
    'process.env.NODE_ENV': '"production"'
})`
`new webpack.DefinePlugin({
    'process.env.BABEL_ENV': '"production"'
})`
Would be valid as well, what do you think of this approach?

Since react defaults to development mode when you import it using npm, Facebook recommends setting the process.env.NODE_ENV property to a ‘production' String + using a minifier tool such as UglifyJS in order to remove all that dead code from our final bundle.
That's because every good minifier tool  would strip values such as

```
if ("production" === "development") {
  console.log('development only')
}
```

from our bundle.
See https://facebook.github.io/react/downloads.html.